codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -282,43 +256,83 @@ public void writePendingBytes() throws Http2Exception {
     /**
      * The remote flow control state for a single stream.
      */
-    private final class DefaultState extends AbstractState {
+    private final class FlowState implements StreamByteDistributor.StreamState {
+        private final Http2Stream stream;
         private final Deque<FlowControlled> pendingWriteQueue;
         private int window;
         private int pendingBytes;
-        // Set to true while a frame is being written, false otherwise.
+        private boolean markedWritable;
+
+        /**
+         * Set to true while a frame is being written, false otherwise.
+         */
         private boolean writing;
-        // Set to true if cancel() was called.
+        /**
+         * Set to true if cancel() was called.
+         */
         private boolean cancelled;
+        private BooleanSupplier isWritableSupplier = new BooleanSupplier() {
+            @Override
+            public boolean get() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) A getX() method which returns a boolean should be named isX() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ABooleanGetMethodName)